pkg/client/restclient/config.go

```diff
@@ -62,6 +62,9 @@ type Config struct {
 	// TODO: demonstrate an OAuth2 compatible client.
 	BearerToken string
 
+	// Server requires plugin-specified authentication.
+	AuthPlugin string
```

Comment:
My initial (self-serving) thought was that other plugins could handle config themselves. E.g. @jdef's DCOS plugin could read an environment variable, another plugin could have its own conf file.

That said, I am very open to changes in structure/naming/etc.